packages/drivers/iframe-driver/src/outerDocumentServiceFactory.ts

```diff
@@ -102,51 +119,72 @@ export class DocumentServiceFactoryProxy implements IDocumentServiceFactoryProxy
         const clientId = deltaStream.clientId;
         const combinedDriver = {
             clientId,
+            getResolvedUrl: Comlink.proxy(async () => resolvedUrl),
             stream: this.getOuterDocumentDeltaConnection(deltaStream),
             deltaStorage: this.getDeltaStorage(deltaStorage),
             storage: this.getStorage(storage),
             logger: this.getLogger(outerProxyLogger),
         };
 
-        this.clients[clientId] = combinedDriver;
+        this._clients[clientId] = combinedDriver;
 
         return clientId;
     }
 
     // TODO: Issue-2109 Implement detach container api or put appropriate comment.
+    // Currently this is mostly copied from RouterliciousDocumentServiceFactory
     public async createContainer(
```

Comment:
i don't understand why anything is copied, shouldn't this just be using the real documentServiceFactory that is passed in?

Reply:
this shouldn't be assuming anything about the server. it should used the public interfaces on our existing driver apis. if that doesn't work, we should add new apis or modify existing apis

---
In reply to: [527056264](https://github.com/microsoft/FluidFramework/pull/4404#discussion_r527056264) [](ancestors = 527056264)